src/test/java/io/debezium/operator/dependent/ServiceAccountDependentTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.debezium.operator.dependent;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.debezium.operator.DebeziumServer;
+import io.debezium.operator.DebeziumServerReconciler;
+import io.fabric8.kubernetes.api.model.ObjectMetaBuilder;
+import io.fabric8.kubernetes.api.model.ServiceAccount;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.javaoperatorsdk.operator.ReconcilerUtils;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.quarkus.kubernetes.client.runtime.KubernetesClientUtils;
+
+public class ServiceAccountDependentTest {
+
+    private final String dsName = "test-ds";
+    private final String saName = "test-ds-service-account";
+    private final String managedSaName = "test-ds-sa";
+    private final String namespace = "debezium";
+    private final KubernetesClient client = KubernetesClientUtils.createClient();
+    private Context<DebeziumServer> context;
+    private DebeziumServer debeziumServer;
+
+    @BeforeEach
+    void before() {
+        debeziumServer = ReconcilerUtils.loadYaml(
+                DebeziumServer.class, DebeziumServerReconciler.class, "ds-postgres-minimal.yml");
+        final var metadata = new ObjectMetaBuilder()
+                .withName(dsName)
+                .withNamespace(namespace)
+                .build();
+        debeziumServer.setMetadata(metadata);
+    }
+
+    @AfterEach
+    void after() {
+        client.resource(debeziumServer).delete();
```

Comment:
@ilyasahsan123  since this is a unit test we are not really creating the object within kubernetes (we shouldn't). So I don't think this is required. Consequently the client field itself is not required.